### PR TITLE
NIAD-963: refactor configuration

### PR DESCRIPTION
**Pull request by n-r-b**
## Description

Refactored configuration to the outcome of this [discussion](https://github.com/nhsconnect/integration-adaptor-lab-results/pull/15#pullrequestreview-578122492), which is to: **Set defaults only in docker-compose, application.yml references docker-compose**. 

This change enables us to manage all the environment variables in one place (docker-compose) so that we don't get confused by defaults hidden elsewhere.

Changes made:

- Removed default values from `main` `application.yml`
- Added missing configuration to `docker-compose.yml`
- Removed majority of environment variables from `intTest` `application.yml`, except from a few which are required by container set up classes
- Removed environment variables from `test` `application.yml`

## Jira Ticket

**AC4 only**

https://gpitbjss.atlassian.net/browse/NIAD-963

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [x] Acceptance Criteria met
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [x] If your pull request depends on any other, please link them in the description
- [x] main branch is passing/green on CI
- [x] Add/update any relevant documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,10 +127,6 @@ A separate source folder [src/intTest](./src/intTest) contains integration tests
     
 ### Debugging
 
-#### MESH API
-
-A `mesh.sh` bash script exists for testing or debugging MESH. For more information see: [mesh/README.md](/mesh/README.md)
```
Comment (n-r-b) [inline README.md:129]: Not part of this task, but on commit it was flagged the link does not exist

**Review by saileshp-kainos — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (andrewwa-kainos) [reply README.md:129]: I'll actually be putting this right back in again for NIAD-957, so perhaps keep it?

Comment (n-r-b) [reply README.md:129]: Sure - i'll put this back then 👍

Comment (andrewwa-kainos) [reply README.md]: TYVM!

**Review by andrewwa-kainos — APPROVED** [review]
LGTM

**Review by saileshp-kainos — APPROVED** [review]

**Pull request merged**